lib/wibox/widget/background.lua

```diff
@@ -21,8 +22,17 @@ function background:draw(context, cr, width, height)
         return
     end
 
+    -- Keep the shape path in case there is a border
+    self._path = nil
+
     if self._shape then
-        self._shape(cr, width, height, unpack(self._shape_args or {}))
+        -- Only add the offset if there is something to draw
+        local offset = ((self._shape_border_width and self._shape_border_color)
+            and self._shape_border_width or 0) / 2
+
+        cr:translate(offset, offset)
+        self._shape(cr, width - 2*offset, height - 2*offset, unpack(self._shape_args or {}))
```

Comment:
Here is my test (together with a proposed fix):
```diff
diff --git a/awesomerc.lua b/awesomerc.lua
index 5fb42d3..71a8d81 100755
--- a/awesomerc.lua
+++ b/awesomerc.lua
@@ -123,6 +123,10 @@ mymainmenu = awful.menu({ items = { { "awesome", myawesomemenu, beautiful.awesom
 
 mylauncher = awful.widget.launcher({ image = beautiful.awesome_icon,
                                      menu = mymainmenu })
+mylauncher = wibox.widget.background(mylauncher)
+mylauncher:set_shape(gears.shape.isosceles_triangle)
+mylauncher:set_shape_border_width(2)
+mylauncher:set_shape_border_color("#ff0000")
 
 -- Menubar configuration
 menubar.utils.terminal = terminal -- Set the terminal for applications that require it
diff --git a/lib/wibox/widget/background.lua b/lib/wibox/widget/background.lua
index 46cb487..6e41269 100644
--- a/lib/wibox/widget/background.lua
+++ b/lib/wibox/widget/background.lua
@@ -17,6 +17,10 @@ local unpack = unpack or table.unpack
 
 local background = { mt = {} }
 
+local function p(a)
+    return a and (a.x .. "," .. a.y)
+end
+
 --- Draw this widget
 function background:draw(context, cr, width, height)
     if not self.widget or not self.widget.visible then
@@ -33,7 +37,12 @@ function background:draw(context, cr, width, height)
 
         cr:translate(offset, offset)
         self._shape(cr, width - 2*offset, height - 2*offset, unpack(self._shape_args or {}))
+        cr:translate(-offset, -offset)
         self._path = cr:copy_path()
+        for t, pts in self._path:pairs() do
+            print("path has", t, p(pts[1]), p(pts[2]), p(pts[3]))
+        end
+        print("done", "btw size is", width, height, "with offset", offset)
         cr:clip()
     end
 
```
Without the new `cr:translate()` this prints:
```
path has	MOVE_TO	9,0	nil	nil
path has	LINE_TO	18,18	nil	nil
path has	LINE_TO	0,18	nil	nil
path has	CLOSE_PATH	nil	nil	nil
path has	MOVE_TO	9,0	nil	nil
done	btw size is	20	20	with offset	1
```
With the new `cr:translate()` this becomes:
```
path has	MOVE_TO	10,1	nil	nil
path has	LINE_TO	19,19	nil	nil
path has	LINE_TO	1,19	nil	nil
path has	CLOSE_PATH	nil	nil	nil
path has	MOVE_TO	10,1	nil	nil
done	btw size is	20	20	with offset	1
```
(Ignore the extra `MOVE_TO` at the end, that's something that `cr:close_path()` does internally)

So could you also add the extra `cr:translate()` that my patch above shows? Thanks. With that fixed, this PR can be merged.